src/less/radio/base/radio.less

```diff
@@ -2,89 +2,61 @@
     display: inline-flex;
     font-size: @font-size-16;
     vertical-align: text-bottom;
-
-    &__checked {
-        color: @radio-checked-color;
-    }
 }
 
-.radio__icon {
+span.radio__icon {
     display: inline-flex;
     font-size: 1em;
     outline-offset: 1px;
+}
 
-    // progressive enhancement - over-ride hidden SVG
-    &[hidden] {
-        display: inline-flex;
-    }
+// progressive enhancement - override hidden SVG
+span.radio__icon[hidden] {
+    display: inline-flex;
+}
 
-    &:not([hidden]) {
-        .background-icon-base-mixin();
-        .radio-icon-unchecked();
-    }
+svg.radio__checked,
+svg.radio__unchecked {
+    .foreground-icon-base-mixin();
+}
 
-    & svg {
-        .foreground-icon-base-mixin();
+svg.radio__checked {
+    color: @radio-checked-color;
+}
 
-        color: @radio-unchecked-color;
-    }
+svg.radio__unchecked {
+    color: @radio-unchecked-color;
 }
 
-.radio__control[type="radio"] {
+input.radio__control[type="radio"] {
     font-size: 100%;
     margin: 0;
     opacity: 0;
     padding: 0;
     position: absolute;
     z-index: 1;
+}
 
-    &:not(:checked) {
-        & + .radio__icon svg.radio__checked {
-            color: @radio-checked-color;
-            display: none;
-        }
-
-        & + .radio__icon svg.radio__unchecked {
-            color: @radio-unchecked-color;
-            display: inline-block;
-        }
-    }
-
-    &:focus + .radio__icon {
-        outline: 1px dotted #767676;
-    }
-
-    &[disabled] + .radio__icon {
-        opacity: @radio-disabled-opacity;
-    }
-
-    &:checked {
-        & + .radio__icon:not([hidden]) {
-            .background-icon-base-mixin();
-            .radio-icon-checked();
-        }
+input.radio__control[type="radio"] + span.radio__icon svg.radio__checked {
+    display: none;
+}
 
-        & + .radio__icon svg.radio__checked {
-            display: inline-block;
-        }
+input.radio__control[type="radio"] + span.radio__icon svg.radio__unchecked {
+    display: inline-block;
+}
 
-        &:not([disabled]) + .radio__icon svg.radio__checked {
-            color: @radio-checked-color;
-        }
+input.radio__control[type="radio"]:checked + span.radio__icon svg.radio__checked {
+    display: inline-block;
+}
 
-        & + .radio__icon svg.radio__unchecked {
-            color: @radio-unchecked-color;
-            display: none;
-        }
-    }
+input.radio__control[type="radio"]:checked + span.radio__icon svg.radio__unchecked {
+    display: none;
 }
 
-@media screen and (-ms-high-contrast: white-on-black) {
-    .radio__icon:not([hidden]) {
-        .background-image-base64(@radio-unchecked-icon-base64-light);
-    }
+input.radio__control[type="radio"]:focus + span.radio__icon {
+    outline: 1px dotted #767676;
```

Comment:
Should we use a variable here?

Reply:
Yep. I'll be merging in the fix in 9.5.0 once it is released. https://github.com/eBay/skin/pull/926